### PR TITLE
Feature/boss area attack

### PR DESCRIPTION
**Pull request by BNS-MarkUlrich**

### CODE REVIEW

File: Assets/Art/Animations/Boss/Boss_Controller.controller
Comment (swzwij) [inline Assets/Art/Animations/Boss/Boss_Controller.controller:1]: Please use the correct prefix and CamelCase

File: Assets/Art/Animations/Boss/Boss_Controller.controller.meta
Comment (swzwij) [inline Assets/Art/Animations/Boss/Boss_Controller.controller.meta:1]: Please use the correct prefix and CamelCase

File: Assets/Art/Animations/Boss/SK_Boss_Export003.fbx
Comment (swzwij) [inline Assets/Art/Animations/Boss/SK_Boss_Export003.fbx:1]: Please use CamelCase

File: Assets/Art/Animations/Boss/SK_Boss_Export02.fbx
Comment (swzwij) [inline Assets/Art/Animations/Boss/SK_Boss_Export02.fbx:1]: Please use CamelCase

File: Assets/Art/Animations/Boss/SK_Boss_Export_Final.fbx
Comment (swzwij) [inline Assets/Art/Animations/Boss/SK_Boss_Export_Final.fbx:1]: Please use CamelCase

File: Assets/Art/Textures/Boss/DefaultMaterial_baseColor.png
Comment (swzwij) [inline Assets/Art/Textures/Boss/DefaultMaterial_baseColor.png:1]: Please use the correct prefix and CamelCase

File: Assets/Art/Textures/Boss/DefaultMaterial_normal.png
Comment (swzwij) [inline Assets/Art/Textures/Boss/DefaultMaterial_normal.png:1]: Please use the correct prefix and CamelCase

File: Assets/Art/Textures/Boss/DefaultMaterial_occlusionRoughnessMetallic.png
Comment (swzwij) [inline Assets/Art/Textures/Boss/DefaultMaterial_occlusionRoughnessMetallic.png:1]: Please use the correct prefix and CamelCase

File: Assets/Art/Textures/Projector/red-area-HD.png
Comment (swzwij) [inline Assets/Art/Textures/Projector/red-area-HD.png:1]: Please use the correct prefix and CamelCase

File: Assets/PlayerMover.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+using System.Collections;
+using System.Collections.Generic;
+using Unity.VisualScripting;
+using UnityEngine;
+
+public class PlayerMover : MonoBehaviour
```
Comment (swzwij) [inline Assets/PlayerMover.cs]: Missing namespace

File: Assets/PlayerMover.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+{
+    [SerializeField] private LayerMask _obstacleLayers;
+
+    [SerializeField] private float _knockbackDistance;
+    [SerializeField][Range(0.1f, 0.9f)] private float _midPointOffset = 0.5f;
+
+    [SerializeField] private float _rayDistance = 2f; 
+    private void StartMoving(GameObject attacker)
```
Comment (swzwij) [inline Assets/PlayerMover.cs]:
```suggestion
    [SerializeField] private float _rayDistance = 2f; 
    
    private void StartMoving(GameObject attacker)
```

Comment (swzwij) [inline Assets/PlayerMover.cs]: Can be one liner

File: Assets/PlayerMover.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        StartCoroutine(MoveacrossCurve(attacker.transform.position));
+    }
+
+    private IEnumerator MoveacrossCurve(Vector3 attackPosition) 
+    {
+        Vector3 startPoint = transform.position;
+
+        var attackDiraction = attackPosition - startPoint.normalized;
```
Comment (swzwij) [inline Assets/PlayerMover.cs]: Please use explicit type instead of var

File: Assets/Scripts/NPC/Boss.cs [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+            }
+
+            if (TryGetComponent(out _healthData))
+                _healthData.onDie.AddListener(TriggerDie);
+
+            _waypointFollower.OnBossInitialised -= InitBoss;
+        }
+
+        [Server]
+        public override void Interact(NetworkConnection connection, float damageAmount = 0)
```
Comment (swzwij) [inline Assets/Scripts/NPC/Boss.cs]: Can be a one liner

File: Assets/Scripts/NPC/Boss.cs [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+        {
+            _healthData.TakeDamage(damageAmount);
+        }
+
+        public override void PlayInteractingSound()
```
Comment (swzwij) [inline Assets/Scripts/NPC/Boss.cs:101]: Eather remove or add a `//TODO:`

File: Assets/Scripts/NPC/Boss.cs (hunk 4 above)
Comment (swzwij) [inline Assets/Scripts/NPC/Boss.cs]: Missing summary

File: Assets/Scripts/NPC/Boss.cs (hunk 5 above)
Comment (swzwij) [inline Assets/Scripts/NPC/Boss.cs:101]: Missing summary

File: Assets/Scripts/NPC/Boss.cs [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+        {
+            
+        }
+
+        [ObserversRpc]
+        public virtual void BroadcastInteract()
```
Comment (swzwij) [inline Assets/Scripts/NPC/Boss.cs:110]: Missing summary

File: Assets/Scripts/Pathfinding/WaypointFollower.cs [hunk 7]
```diff
@@ -99,10 +131,27 @@ protected IEnumerator WaitForPathClearance()
...
+
             yield return new WaitUntil(() => !IsPathBlocked);
 
             yield return new WaitForSeconds(p_waitTime);
             ContinuePath();
+            OnPathCleared?.Invoke(true);
+        }
+
+        [Server]
+        public void ToggleWaiting(bool isWaiting)
```
Comment (swzwij) [inline Assets/Scripts/Pathfinding/WaypointFollower.cs:132]: Missing summary

**Review by swzwij — CHANGES_REQUESTED** [review]

File: Assets/Art/Animations/Boss/Boss_Controller.controller
Comment (BNS-MarkUlrich) [reply Assets/Art/Animations/Boss/Boss_Controller.controller:1]: There are no prefix conventions for controllers

**Review by swzwij — APPROVED** [review]

**Review by swzwij — APPROVED** [review]

**Pull request merged**